Division by float: convert ProdExpr -> Expr

Dear SCIPers,
Following this https://github.com/SCIP-Interfaces/PySCIPOpt/issues/212#issuecomment-435042087 I am here to ask for a clarification.

Consider:
```python
from pyscipopt import Model
model = Model()
x = model.addVar(vtype="c")
print((2*x).__class__.__name__)
# Expr
print((x/2.).__class__.__name__)
# ProdExpr
print((2.*x).__class__.__name__)
# Expr
print((x/2).__class__.__name__)
# ProdExpr
```
It seems like `ProdExpr` is not casted as `Expr` in any case the division is defined. This provokes:
```python
model.setObjective(.5*x) # accepted
model.setObjective(x/2.) # not accepted
# AssertionError: given coefficient is neither Expr or number but ProdExpr
```
Is this a design choice? Is SCIP behaviour?

As a Python user, I would expected a full support of the division of a continuous variable with a float number.
In other cases like integer variable divided by float or continuous variable divided by integer, I would expect a division btw. floats since integers are casted to floats in these cases.
In case of division between an integer variable and an integer number I would keep the `ProdExpr` until there is no support for division with remainders (that is why I wrote [this](https://github.com/SCIP-Interfaces/PySCIPOpt/pull/211/files#diff-3ff183b9d68ae9718e7ecc814d299d67)).

--- a/src/pyscipopt/expr.pxi
+++ b/src/pyscipopt/expr.pxi
@@ -216,20 +216,32 @@ cdef class Expr:
 
     def __div__(self, other):
         ''' transforms Expr into GenExpr'''
+        if _is_number(other):
+            f = 1.0/float(other)
+            return f * self
         selfexpr = buildGenExprObj(self)
         return selfexpr.__div__(other)
 
     def __rdiv__(self, other):
         ''' other / self '''
+        if _is_number(self):
+            f = 1.0/float(self)
+            return f * other
         otherexpr = buildGenExprObj(other)
         return otherexpr.__div__(self)
 
     def __truediv__(self,other):
+        if _is_number(other):
+            f = 1.0/float(other)
+            return f * self
         selfexpr = buildGenExprObj(self)
         return selfexpr.__truediv__(other)
 
     def __rtruediv__(self, other):
         ''' other / self '''
+        if _is_number(self):
+            f = 1.0/float(self)
+            return f * other
         otherexpr = buildGenExprObj(other)
         return otherexpr.__truediv__(self)
 

--- a/src/pyscipopt/expr.pxi
+++ b/src/pyscipopt/expr.pxi
@@ -545,7 +545,7 @@ cdef class GenExpr:
         divisor = buildGenExprObj(other)
         # we can't divide by 0
         if divisor.getOp() == Operator.const and divisor.number == 0.0:
-            raise ValueError("cannot divide by 0")
+            raise ZeroDivisionError("cannot divide by 0")
         return self * divisor**(-1)
 
     def __rdiv__(self, other):
@@ -557,7 +557,7 @@ cdef class GenExpr:
         divisor = buildGenExprObj(other)
         # we can't divide by 0
         if divisor.getOp() == Operator.const and divisor.number == 0.0:
-            raise ValueError("cannot divide by 0")
+            raise ZeroDivisionError("cannot divide by 0")
         return self * divisor**(-1)
 
     def __rtruediv__(self, other):

--- a/tests/test_expr.py
+++ b/tests/test_expr.py
@@ -51,7 +51,7 @@ def test_upgrade(model):
     assert isinstance(log(expr), GenExpr)
     assert isinstance(exp(expr), GenExpr)
 
-    with pytest.raises(ValueError):
+    with pytest.raises(ZeroDivisionError):
         expr /= 0.0
 
 def test_genexpr_op_expr(model):

